### PR TITLE
Update storage circuits for query ERC20

### PR DESCRIPTION
**Pull request by silathdiir**
***I commented out the tests of block and state circuits.***

### CODE REVIEW

File: mr-plonky2-circuits/src/query_erc20/storage/leaf.rs [hunk 1]
```diff
@@ -1,60 +1,84 @@
...
+        let inputs = prefix
+            .into_iter()
+            .chain(address.arr.map(|v| v.to_target()))
+            .chain(value.arr.map(|v| v.to_target()))
+            .collect();
+        let c = b.hash_n_to_hash_no_pad::<PoseidonHash>(inputs);
+
+        // V = R * value / totalSupply
+        // TODO: U256 operations
+        let v = &reward;
```
Comment (nikkolasg) [inline mr-plonky2-circuits/src/query_erc20/storage/leaf.rs]: v should be v. rewards should be rewards.

File: mr-plonky2-circuits/src/query_erc20/storage/tests.rs [hunk 2]
```diff
@@ -25,393 +34,162 @@ use plonky2::{
...
-        Self::parts_into_values(
-            &mut pis,
-            &root,
-            digest_fs.as_slice().try_into().unwrap(),
-            &leaf_value,
-        );
+    let proof = run_circuit::<_, D, C, _>(test_circuit);
+    let [pi, child_pi] = [&proof.public_inputs, child_pi_slice]
+        .map(|pi| PublicInputs::<GoldilocksField>::from_slice(pi));
+    // TODO: check the hash (p[C])
```
Comment (nikkolasg) [inline mr-plonky2-circuits/src/query_erc20/storage/tests.rs]: can you do it ? would be nice to ensure that the database leaf hash is still the same

**Review by nikkolasg — APPROVED** [review]
One minor request for the test and one misc change, otherwise LGTM

File: mr-plonky2-circuits/src/query_erc20/storage/tests.rs (hunk 2 above)
Comment (silathdiir) [reply mr-plonky2-circuits/src/query_erc20/storage/tests.rs]: Added in commit [bc67b19](https://github.com/Lagrange-Labs/Euclid-database/pull/6/commits/bc67b192a05d040164a52a8ce9ba9bdcf952a573).

File: mr-plonky2-circuits/src/query_erc20/storage/leaf.rs (hunk 1 above)
Comment (silathdiir) [reply mr-plonky2-circuits/src/query_erc20/storage/leaf.rs]: Fixed to `PackedU256Target::new(b)` in commit [bc67b19](https://github.com/Lagrange-Labs/Euclid-database/pull/6/commits/bc67b192a05d040164a52a8ce9ba9bdcf952a573).

**Pull request merged**